### DISCUSSION TITLE
[NextJS] Enhanced Sitecore item scaffolding

### DISCUSSION THREAD

**Discussion by mjarosch** [Sitecore/jss#1397] [Ideas]
## Is your feature request related to a problem? Please describe.

When I am creating new JSS components, I need to write the component props to match the Sitecore item definition and manually keep this in sink.

## Describe the solution you'd like

As a JSS developer, when I am working with Sitecore items, I would like to be able to run a command that reads the serialized Sitecore items and generates Typescript files representing the fields on those items.

## Describe alternatives you've considered

We have tried using a custom template with Leprechaun, but it doesn't work on non-windows operating systems.

**Comment by kendoce** [comment]
@mjarosch - we are considering this as part of our roadmap. It might not be the exact approach but it will accomplish the same result.

**Reply by mjarosch** [reply]
@kendoce Sounds good. We submitted a patch to Leprechaun and have it working for now. One feature that would be nice, which can't be done with Leprechaun right now, is generate constants (item ids, field ids, etc) from the items as well.
